src/CatalogService/src/Application/Product/Commands/AddProduct/AddProductCommand.cs

```diff
@@ -1,47 +1,46 @@
-﻿using CatalogService.Application.Common.Interfaces;
-using MediatR;
-
-namespace CatalogService.Application.Product.Commands.AddProduct;
-
-public class AddProductCommand : IRequest<int>
-{
-    public string Name { get; set; }
-
-    public string Description { get; set; }
-
-    public int Image { get; set; }
-
-    public int CatalogId { get; set; }
-
-    public decimal Price { get; set; }
-
-    public uint Amount { get; private set; }
-}
-
-public class AddProductCommandHandler : IRequestHandler<AddProductCommand, int>
-{
-    private readonly IApplicationDbContext _applicationDbContext;
-
-    public AddProductCommandHandler(IApplicationDbContext applicationDbContext)
-    {
-        _applicationDbContext = applicationDbContext;
-    }
-
-    public async Task<int> Handle(AddProductCommand request, CancellationToken cancellationToken)
-    {
-        var product = new Domain.Entities.Product()
-        {
-            Name = request.Name,
-            Description = request.Description,
-            Image = request.Image,
-            CatalogId = request.CatalogId,
-            Price = request.Price,
-            Amount = request.Amount
-        };
-
-        _applicationDbContext.Products.Add(product);
-        await _applicationDbContext.SaveChangesAsync(cancellationToken);
-
-        return product.Id;
-    }
-}
+﻿using CatalogService.Application.Common.Interfaces;
+using MediatR;
+
+namespace CatalogService.Application.Product.Commands.AddProduct;
+
+public class AddProductCommand : IRequest<int>
+{
+    public string Name { get; set; }
+
+    public string Description { get; set; }
+
+    public int Image { get; set; }
+
+    public int CatalogId { get; set; }
+
+    public decimal Price { get; set; }
+
+    public uint Amount { get; private set; }

```

Comment:
I think it's now broken. How would Amount be populated if it has a private set?
Not sure if deserialization would work here properly.

Reply:
It will be populated. there is a backing field and EF is set up to use it for "Amount" - ProductConfiguration